File: tests/0020_valid_parentheses_test.cpp
#include "0020_valid_parentheses.cpp"
#include <doctest.h>

TEST_CASE("ValidParenthesesTests")
{
    Solution s = Solution();
    CHECK(s.isValid("()"));
    CHECK(s.isValid("()[]{}"));
    CHECK_FALSE(s.isValid("(]"));
    CHECK_FALSE(s.isValid("([)]"));
    CHECK(s.isValid("{[]}"));
    CHECK(s.isValid(""));
    CHECK_FALSE(s.isValid("}"));
    CHECK_FALSE(s.isValid("(])"));
}

File: tests/0020_valid_parentheses_test.py
import importlib

module = importlib.import_module("algorithms.0020_valid_parentheses")


def test_valid_parentheses():
    s = module.Solution()
    assert s.isValid("()")
    assert s.isValid("()[]{}")
    assert not s.isValid("(]")
    assert not s.isValid("([)]")
    assert s.isValid("{[]}")
    assert s.isValid("")
    assert not s.isValid("}")
    assert not s.isValid("(])")
